### ISSUE TITLE
Official Dart lint rules applied with recommend set

### ISSUE THREAD

**Issue by navibyte** [navibyte/dataflow#2]
BETA version 0.6.0 introduced custom lint rules, see [Lint rules and analysis options updated #8](https://github.com/navibyte/geospatial/issues/8) from geospatial repo. 

However since that the Dart team has announced new official lint rules:
* [Official Dart lint rules](https://pub.dev/packages/lints)
* [Customizing static analysis](https://dart.dev/guides/language/analysis-options)
* [Official Flutter lint rules](https://pub.dev/packages/flutter_lints)

In general, it's better to follow recommend lints to keep code consistent with other developers.

Steps to configure official lints in old package
1. First apply `dart pub add --dev lints` that adds `lints: ^1.0.1` or other version to `dev_dependencies`
2. Then edit `analysis_options.yaml` by adding `include: package:lints/recommended.yaml`
3. Customize `analysis_options.yaml`

**Comment by navibyte** [comment]
Modified lint rules and analysis_options.yaml as follows - testing these:

```yaml
# Analysis options / lint rules (2021-07-10)
# 
# Documentation (official for Dart)
#     https://pub.dev/packages/lints
#     https://dart.dev/guides/language/analysis-options
#     https://dart-lang.github.io/linter/lints/
#
# Documentation (official for Flutter)
#     https://pub.dev/packages/flutter_lints
#
# Getting started (before new official Dart and Flutter lints announced!)
#     https://dash-overflow.net/articles/getting_started/
#         Some of comments originated from this blog.
#     https://rydmike.com/blog => My Flutter Linting Preferences (Jan 10, 2021)
#         Some of comments originated from this blog.

## Include recommended set of the Official Dart lint rules.
include: package:lints/recommended.yaml

analyzer:
  exclude:
    # Ignore warnings of files generated by json_serializable, built_value etc.
    - "**/*.g.dart"

    # Ignore warnings of files generated by Freezed.
    - "**/*.freezed.dart"

  language:
    # See => https://github.com/dart-lang/language/blob/master/resources/type-system/strict-raw-types.md
    strict-raw-types: true

  strong-mode:
    # Explained best here => https://dash-overflow.net/articles/getting_started/
    implicit-casts: false

    # Explained best here => https://dash-overflow.net/articles/getting_started/
    implicit-dynamic: false
```

**Comment by navibyte** [comment]
Implemented in BETA-version 0.7.0
